concordium-std-derive/src/lib.rs

```diff
@@ -1939,6 +1943,113 @@ fn concordium_test_worker(_attr: TokenStream, item: TokenStream) -> syn::Result<
     Ok(test_fn.into())
 }
 
+/// Look up the `tests` identifier in `NestedMeta` and return the value
+/// associated with it. If no `tests` is found or parsing the value has failed,
+/// return a error
+#[cfg(feature = "concordium-quickcheck")]
+fn get_quickcheck_tests_count(meta: &NestedMeta) -> Result<u64, syn::parse::Error> {
+    match meta {
+        NestedMeta::Meta(Meta::NameValue(v)) => {
+            if v.path.is_ident("tests") {
+                match &v.lit {
+                    Lit::Int(i) => i
+                        .base10_parse::<u64>()
+                        .map_err(|_| syn::parse::Error::new(i.span(), "error when pasing integer")),
+                    _ => Err(syn::parse::Error::new(v.span(), "unexpected attribute")),
+                }
+            } else {
+                Err(syn::parse::Error::new(v.span(), "unexpected attribute"))
+            }
+        }
+        NestedMeta::Meta(m) => Err(syn::parse::Error::new(m.span(), "unexpected attribute")),
+        NestedMeta::Lit(l) => {
+            Err(syn::parse::Error::new(l.span(), "expected a named attribute, e.g. tests = 1000"))
+        }
+    }
+}
+
+/// A default number of tests to run with QuickCheck
+#[cfg(feature = "concordium-quickcheck")]
+const DEFAULT_QUICKCHECK_TESTS: u64 = 100;
+
+#[cfg(feature = "concordium-quickcheck")]
+#[proc_macro_attribute]
+/// Derive the appropriate export for an annotated QuickCheck function by
+/// exposing it as `#[concordium_test]`. The macro was copied from `QuickCheck`
+/// and changed to use `concordium_std::test_infrastructure::concordium_qc`
+/// instead of the standard  `QuickCheck`'s `quickcheck`
+///
+/// The macro takes a `tests` attrubute that specifies how many tests to run:
+/// `#[concordium_quickcheck(tests = 1000)]`. If no `tests` is provided,
+/// `DEFAULT_QUICKCHECK_TESTS` is used
+pub fn concordium_quickcheck(attr: TokenStream, input: TokenStream) -> TokenStream {
+    use syn::AttributeArgs;
+
+    let output = match syn::Item::parse.parse(input.clone()) {
```

Comment:
I think you can avoid cloning
By converting the tokenstream to a `proc_macro2::TokenStream` and then parse using [`parse2`](https://docs.rs/syn/latest/syn/fn.parse2.html)

Reply:
I can't figure out how :) I think `attr` is consumed when parsing, and then I need `span` to report an error.

Reply:
You could get the span before calling parse2
```rust
pub fn concordium_quickcheck(_attr: TokenStream, input: TokenStream) -> TokenStream {
    let input = proc_macro2::TokenStream::from(input);
    let span = input.span();
    let output = match syn::Item::parse.parse2(input) {
        Ok(syn::Item::Fn(mut item_fn)) => {/*...*/}
        _ => {
            let msg = "#[concordium_quickcheck] can only be applied to functions.";
            syn::parse::Error::new(span, msg).to_compile_error()
        }
    }
    /*...*/
}
```

Reply:
Right, this works, indeed, thanks!